### PR TITLE
Scripts and artifacts to create releases

### PR DESCRIPTION
**Pull request by csrwng**
Initial commit of code to create releases for the ibm-roks-toolkit
This requires configuration changes on the CI side to take effect.

### CODE REVIEW

File: hack/create-git-release.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+fi
+
+RELEASE="$(cat "${REPODIR}/release")"
+RELEASE_DATE="$(cat "${REPODIR}/release-date")"
+RELEASE_TAG="v${RELEASE}-${RELEASE_DATE}"
+RELEASE_BRANCH="release-${RELEASE}"
+
+# if the branch corresponding to the release doesn't exist, skip because 
+# it's likely the master branch
+if ! git rev-parse --verify "${RELEASE_BRANCH}" &> /dev/null; then
```
Comment (rtheis) [inline hack/create-git-release.sh:20]: We should probably verify that we only create releases from the matching branch.

File: hack/create-git-release.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+if [[ ! -f "${REPODIR}/release" ]]; then
+  echo "Release name file (release) does not exist. Nothing to do"
+fi
+if [[ ! -f "${REPODIR}/release-date" ]]; then
+  echo "Release date file (release-date) does not exist. Nothing to do"
+fi
+
+RELEASE="$(cat "${REPODIR}/release")"
+RELEASE_DATE="$(cat "${REPODIR}/release-date")"
+RELEASE_TAG="v${RELEASE}-${RELEASE_DATE}"
```
Comment (rtheis) [inline hack/create-git-release.sh:15]: We are currently using `+` not `-`.  Although we are using `-` for the image tags.

image

**Review by rtheis — CHANGES_REQUESTED** [review]
It appears that we are changing the release artifact names.  I want to ensure this was intended before proceeding since I suspect we will need to update our internal tooling.

File: hack/create-git-release.sh (hunk 2 above)
Comment (csrwng) [reply hack/create-git-release.sh:15]: @rtheis thank you for catching that. I will fix

File: hack/create-git-release.sh (hunk 1 above)
Comment (csrwng) [reply hack/create-git-release.sh:20]: Added a check to ensure that the branch name corresponds to the expected release branch

File: hack/create-git-release.sh (hunk 2 above)
Comment (csrwng) [reply hack/create-git-release.sh]: Fixed

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+For every PR that is merged into this repository, a post-commit CI job checks whether a release should be created. If so,
+it will create a new release by publishing release artifacts to the git repository and tagging images in the CI registry
+with the release tag.
+
+Releases are controlled by 2 files in the root of the repository:
+
+- release - Contains the major.minor version of the ibm-roks-toolkit branch. This should remain constant and only change
+  when a new release branch is created.
+- release-date - The release date in yyyymmdd format. This is used to create the release tag which is in the format:
+  `v[RELEASE]-[RELEASE_DATE]`
```
Comment (rtheis) [inline RELEASES.md:13]: nit: `v[RELEASE]-[RELEASE_DATE]` should be `v[RELEASE]+[RELEASE_DATE]`

**Review by rtheis — APPROVED** [review]
/lgtm
/approved

File: RELEASES.md (hunk 3 above)
Comment (csrwng) [reply RELEASES.md:13]: Thanks! https://github.com/openshift/ibm-roks-toolkit/pull/398

**Pull request merged**